index.js

```diff
@@ -69,23 +76,35 @@ function RequestStream(options) {
   options = options || {};
   if (!options.baseurl) throw new Error('options.baseurl should be an http:// or https:// baseurl for replay requests');
   if (!options.hwm) options.hwm = 100;
-
-  function transform(pathname, enc, callback) {
+  function transform(data, enc, callback) {
     if (this._closed) return setImmediate(callback);
-
+    var pathname, referer;
+    if (typeof data === 'object') {
+      pathname = data[0];
+      referer = data[1];
+      if (referer && typeof referer !== 'string') referer = referer.toString('utf8');
```

Comment:
Ooh interesting! Out of curiosity, in what scenario is `referer` not a string?

Reply:
I think there may be some cases where data comes in as a buffer. I added this line because we're doing it with `pathname` too here https://github.com/mapbox/aws-log-replay/blob/master/index.js#L76